spec/BowlingGameSpec.js

```diff
@@ -0,0 +1,97 @@
+describe('bowling game logic', function(){
+  var game;
+
+  beforeEach(function(){
+    game = new BowlingGame();
+  });
+
+  it('starts with score = 0', function() {
+    expect(game.score).toEqual(0);
+  });
+
+  it('has 10 frames', function() {
+    expect(game.allFrames.length).toEqual(10);
+  });
+
+  it('can roll a ball', function() {
+    game.roll(2);
+    expect(game.score).toEqual(2);
+  });
+
+  it('can only take rolls that are 10 or less', function() {
+    expect(function() { game.roll(11) } ).toThrow(new Error ("Illegal throw"));
+    expect(game.score).toEqual(0);
+  })
+
+  it('a roll in input into the frame scores', function() {
+    game.roll(5);
+    game.roll(4);
+    expect(game.allRolls[0]).toEqual([5,4]);
+  });
+
+  it('can only roll twice before moving onto another frame', function() {
+    game.roll(2);
+    game.roll(2);
+    game.roll(2);
+    game.roll(2);
+    expect(game.allRolls[0]).toEqual([2,2]);
+    expect(game.allRolls[1]).toEqual([2,2]);
+    expect(game.score).toEqual(8);
+  });
+
+  it('only rolls once if a strike is scored', function() {
+  game.roll(10);
+  game.roll(5);
+  expect(game.allFrames[0]).toEqual(10);
```

Comment:
Again, breaking encapsulation here. You could have a method like `game.getFrame(0)`;